pkg/deployment/collector_test.go

```diff
@@ -488,6 +488,21 @@ func TestAutoscalersDisabledByExplicitOption(t *testing.T) {
 	assert.Len(t, a, 0)
 }
 
+func TestAutoscalersSetMaxReplicas(t *testing.T) {
```

Comment:
I don't understand what is going on. Could you please explain this test and is the name of the test correct?

Does the autoscaler set the number of replicas? The replicas are set directly in the CR and then autoscaler has the same value.

Reply:
This test is covering a hole in coverage that I found when working on this PR. Basically, when `MaxReplicas` is set in our CR, we pass it along to the HPA's `MaxReplicas` value. If we don't have an explicit value, we set an hard upper bound, `100`.

Reply:
Shouldn't this func test all replica fields `Replicas`, `MinReplicas` and `MaxReplicas`? 

It would help me to understand the behavior if we had all combinations in the single test via:

```
tests := []struct{
```
replicas int,
minReplicas int,
maxReplicas int,
//...
//expected values for these fields
}

Reply:
Sorry, I should have mentioned that the behavior I described is the `Deployment`'s and `HPA`'s. With our operator as part of the mix, setting an explicit value for `Replicas` will disable `HPA`, as we understand that people would want a fixed amount of replicas. I just checked and we have a test describing this case (`TestAutoscalersDisabledByExplicitReplicaSize`)